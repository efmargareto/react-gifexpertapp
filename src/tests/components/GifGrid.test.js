import { shallow } from 'enzyme';
import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import '@testing-library/jest-dom'
import { useFetchGifs } from '../../hooks/useFecthGifs';
jest.mock('../../hooks/useFecthGifs')

describe('Pruebas en componente <GifGrid />', () => {
    
    const category = 'Wow'

    test('Debe hacer match con el snapShot ', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow( <GifGrid category = { category } /> )

        expect( wrapper ).toMatchSnapshot()
    })

    test('Debe de mostrar items cuando se cargan imagenes', () => {

        const gifs = [{
            id: 'Anguel',
            url: 'https//localhost/img/anguel.jpg',
            title: 'Titulo'
        }] 

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        
        const wrapper = shallow( <GifGrid category = { category } /> )

        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('p').exists() ).toBe(false)
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )
    })
    
})
