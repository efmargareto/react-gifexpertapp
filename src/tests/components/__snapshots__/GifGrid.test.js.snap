// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifGrid /> Debe de mostrar items cuando se cargan imagenes 1`] = `
<Fragment>
  <h3>
     
    Wow
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="Anguel"
      img={
        Object {
          "id": "Anguel",
          "title": "Titulo",
          "url": "https//localhost/img/anguel.jpg",
        }
      }
      key="Anguel"
      title="Titulo"
      url="https//localhost/img/anguel.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en componente <GifGrid /> Debe hacer match con el snapShot  1`] = `
<Fragment>
  <h3>
     
    Wow
     
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
